include/buffers.h

```diff
@@ -5,13 +5,31 @@
 #include "macros.h"
 #include "stack.h"
 
+// pre-boot variables
+extern u16 gFramebuffer1[SCREEN_HEIGHT][SCREEN_WIDTH]; // at 0x80000500
+extern u8 gPreBootBuffer[];
+
 
 extern u8 gGfxSPTaskYieldBuffer[OS_YIELD_DATA_SIZE];
 extern STACK(gGfxSPTaskStack, 0x400);
 extern GfxPool gGfxPools[2];
 extern u8 gAudioHeap[0x138000];
 extern u8 gSystemHeap[];
 
+/**
+ * The system heap is an area of memory that covers all the RAM that is not
+ * taken by compiled code/data.
+ * It starts just after the end of compiled code/data (that is not an overlay)
+ * and ends at a fixed location in RAM.
+ * That fixed location in RAM is used by the framebuffers of the game and have
+ * a fixed size.
+ * This address is calculated by doing `0x80800000 - (size of framebuffers)`,
+ * where 0x80800000 is the end of the Expansion Pak address range.
```

Comment:
Could we possibly frame the define like this comment, something like
`#define SYSTEM_HEAP_END_ADDR 0x80800000 - (size of framebuffers)`, though probably want 0x80800000 like FAULT_FB_ADDRESS and do `PHYS_TO_K0(0x800000)` instead.

Also not sure but would it be better to have the define be something like `FB_ADDRESS` instead and phrase it in terms of the framebuffers, rather than the end of something.

Reply:
I named it as the end of the system_heap segment because that's how it is used on `Main`:
```
    sysHeap = (intptr_t)SEGMENT_START(system_heap);
    fb = SYSTEM_HEAP_END_ADDR;
    gSystemHeapSize = fb - sysHeap;
    SystemHeap_Init((void*)sysHeap, gSystemHeapSize);
```

Reply:
Yeah, though you could see it as the beginning of the framebuffers (which is why it is the end of the system heap), which is actually how it currently portrayed with the temp variable being named `fb`.